src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -636,12 +639,19 @@ std::unique_ptr<data_iterator> activity_classifier::create_iterator(
   data_params.prediction_window = read_state<flex_int>("prediction_window");
   data_params.predictions_in_chunk = NUM_PREDICTIONS_PER_CHUNK;
   data_params.use_data_augmentation = use_data_augmentation;
-  data_params.random_seed = read_state<int>("random_seed");
+  if (!use_data_augmentation) {
+    data_params.random_seed = 0;
+  } else {
+    data_params.random_seed = read_state<int>("random_seed");
+  }
   return std::unique_ptr<data_iterator>(new simple_data_iterator(data_params));
 }
 
 std::unique_ptr<compute_context> activity_classifier::create_compute_context()
     const {
+  if (state.find("use_tensorflow") == state.end()){
```

Comment:
I think we want to simplify this logic, not extend it. Again, please sync up with Shreya and Fareeha here